pkg/retry/retry_test.go

```diff
@@ -0,0 +1,83 @@
+package retry
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	goretry "github.com/sethvargo/go-retry"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewNoOpRetryer(t *testing.T) {
+	retryer := NewNoOpRetryer()
+	require.NotNil(t, retryer)
+	require.NotNil(t, retryer.config)
+	require.NotNil(t, retryer.config.BackoffStrategy)
+
+	expectedBackoffStrategy := goretry.NewConstant(time.Second * 1)
+	expectedBackoffStrategy = goretry.WithMaxRetries(0, expectedBackoffStrategy)
+
+	require.IsType(t, expectedBackoffStrategy, retryer.config.BackoffStrategy)
+}
+
+func TestDefaultBackoffStrategy(t *testing.T) {
+	backoff := DefaultBackoffStrategy()
+	require.NotNil(t, backoff)
+}
+
+func TestNewDefaultRetryer(t *testing.T) {
+	retryer := NewDefaultRetryer()
+	require.NotNil(t, retryer)
+	require.NotNil(t, retryer.config)
+	require.NotNil(t, retryer.config.BackoffStrategy)
+
+	expectedBackoffStrategy := goretry.NewConstant(time.Second * 1)
+	expectedBackoffStrategy = goretry.WithMaxRetries(0, expectedBackoffStrategy)
+
+	require.IsType(t, expectedBackoffStrategy, retryer.config.BackoffStrategy)
+}
+
+func TestNewRetryer(t *testing.T) {
+	config := &RetryConfig{
+		BackoffStrategy: goretry.NewConstant(1 * time.Second),
+	}
```

Comment:
We can test this with a **RetryConfig** that has a nil **BackOffStrategy**.

Reply:
added a test